src/bvh/bvh_impl.rs

```diff
@@ -107,7 +107,17 @@ impl<T: BHValue, const D: usize> Bvh<T, D> {
             return Vec::new();
         }
         let mut indices = Vec::new();
-        BvhNode::traverse_recursive(&self.nodes, 0, ray, &mut indices);
+        // Manually handle the case of a single-node BVH, because `traverse_recursive`
+        // cannot access the root shape's AABB.
```

Comment:
We _could_ change this I suppose and give `traverse_recursive()` a reference to the root _in theory_. I'm not sure it would be cleaner in the end. What do you think? We could even give it `&self` and make the `BvhNode` carry a reference to the root (though this would probably be a large change and I'm not sure it's even useful).

Reply:
I would have done that except for the fact that `traverse_recursive` is public-API. AFAIK it shouldn't be. If you're fine with privatizing it, I'd make this change.

Reply:
That's a good point. I don't see a reason people should use `traverse_recursive` directly over `traverse`.